addons/print/models/account_invoice.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+from openerp import api, models
+from openerp.tools.translate import _
+
+
+class account_invoice(models.Model):
+    """ Printable account_invoice.
+    """
+
+    _name = 'account.invoice'
+    _inherit = ['account.invoice', 'print.mixin']
+
+    def print_validate_sending(self):
+        super(account_invoice, self).print_validate_sending()
+        PrintOrder = self.env['print.order']
+        for record in self:
+            order = PrintOrder.search([('res_model', '=', 'account.invoice'), ('res_id', '=', record.id)], limit=1, order='send_date desc')
+            if order:
+                # put confirmation message in the chatter
+                message = _("This invoice was sent by post with the provider <i>%s</i> at the following address. <br/><br/> \
```

Comment:
Here you may use a dict and/or format to ease the reading of the string formatting.